src/common/FileSystem.cpp

```diff
@@ -1059,8 +1056,20 @@ static void InternalLoadPak(const PakInfo& pak, Util::optional<uint32_t> expecte
 			return;
 	}
 
-	// Add the pak to the list of loaded paks
-	fsLogs.Notice("Loading pak '%s'...", pak.path.c_str());
+	if (pak.type == pakType_t::PAK_ZIP) {
+		if (!isLegacy) {
+			fsLogs.Notice("Loading pak '%s'...", pak.path.c_str());
+		} else {
+			fsLogs.Notice("Loading legacy pak '%s'...", pak.path.c_str());
+		}
+	} else if (pak.type == pakType_t::PAK_DIR) {
+		if (!isLegacy) {
+			fsLogs.Notice("Loading pakdir '%s'...", pak.path.c_str());
+		} else {
+			fsLogs.Notice("Loading legacy pakdir '%s'...", pak.path.c_str());
+		}
+	}
```

Comment:
There are two choices, either test all cases in else-ifs and add an else clause with UNREACHABLE, or instead of finishing with else if (some condition) {code ...} do else {ASSERT(some condition), code...}. This way if a new case arises, we'll get a warning and know that things should be fixed.